### PR TITLE
One FileKey per File transition, FileKey no longer needed

### PR DESCRIPTION
**Pull request by Hirobreak**
- Migration to remove filekey and move keys and ivs to file table
- Using key and iv in files metadata to encrypt and decrypt

### CODE REVIEW

File: electron_app/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   },
   "scripts": {
     "electron": "electron .",
-    "set-env": "NODE_ENV=development MAILBOX_URL=http://localhost:3000 DIALOG_URL=http://localhost:3006 LOGIN_URL=http://localhost:3005 LOADING_URL=http://localhost:3003 COMPOSER_URL=http://localhost:3004",
+    "set-env": "DEBUG=knex:*  NODE_ENV=development MAILBOX_URL=http://localhost:3000 DIALOG_URL=http://localhost:3006 LOGIN_URL=http://localhost:3005 LOADING_URL=http://localhost:3003 COMPOSER_URL=http://localhost:3004",
```
Comment (erikaperugachi) [inline electron_app/package.json]: delete knex*

File: electron_app/src/ipc/database.js [hunk 2]
```diff
@@ -153,10 +153,6 @@ ipc.answerRenderer('db-get-files-by-tokens', tokens =>
   dbManager.getFilesByTokens(tokens)
 );
 
-ipc.answerRenderer('db-get-filekey-by-emailid', emailId =>
```
Comment (erikaperugachi) [inline electron_app/src/ipc/database.js]: delete call

File: electron_app/src/migrations/20190111104210_dropFileKey.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+const createColumn = knex => {
+  return knex.schema.table(Table.FILE, table => {
+    table.string('key', MEDIUM_STRING_SIZE);
+    table.string('iv', MEDIUM_STRING_SIZE);
+  });
+};
+
+const setKeyIvToFile = async knex => {
+  return await knex.transaction(async trx => {
+    const fileKeys = await trx(Table.FILE_KEY).select();
```
Comment (JulianAdams4) [inline electron_app/src/migrations/20190111104210_dropFileKey.js]: Check `select` in batches

File: electron_app/src/migrations/20190111104210_dropFileKey.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    FILE_KEY_AFTER_DELETE_EMAIL,
+    Table.FILE_KEY
+  );
+};
+
+const reFillFileKey = async knex => {
+  return await knex.transaction(async trx => {
+    const files = await trx(Table.FILE)
+      .distinc('emailId')
+      .select();
```
Comment (JulianAdams4) [inline electron_app/src/migrations/20190111104210_dropFileKey.js]: Same

**Pull request merged**